### PR TITLE
groupadd, groupmod: Update gshadow file with -U

### PR DESCRIPTION
**Pull request by sandwm**
When running groupadd or groupmod with the -U|--user option, also update the group shadow database if it is used.

This closes <https://github.com/shadow-maint/shadow/issues/1124>. See there for more information.

CC @alejandro-colomar

### CODE REVIEW

File: src/groupadd.c
Comment (alejandro-colomar) [inline src/groupadd.c:210]: Please remove the braces for a one-liner.

File: src/groupmod.c
Comment (alejandro-colomar) [inline src/groupmod.c:271]: Please remove the whitespace before the `(`.

Comment (sandwm) [reply src/groupmod.c]: Done. Note: That superfluous whitespace is also present in several other function calls, e.g. the equivalent one in line 262.

File: src/groupadd.c
Comment (sandwm) [reply src/groupadd.c]: Done.

File: src/groupmod.c
Comment (alejandro-colomar) [reply src/groupmod.c]: Yep, that's something I'm working on.  Slowly.  :)

File: src/groupmod.c
Comment (alejandro-colomar) [inline src/groupmod.c:286]: Please remove braces.

**Review by hallyn — APPROVED** [review]
It looks good to me.  Would like someone else to take a close look before merging.

**Comment by alejandro-colomar** [comment]
@ikerexxe , would you mind looking at this?  :)

**Review by alejandro-colomar — CHANGES_REQUESTED** [review]
Since @ikerexxe cannot reproduce the bug, let's hold on until after the release to clarify.

**Review by ikerexxe — APPROVED** [review]
LGTM! Thank you for taking care of it.

**Comment by ikerexxe** [comment]
@alejandro-colomar do you mind removing your change request? It blocks the merge

**Comment by alejandro-colomar** [comment]
Thanks!  I'll merge now.

**Pull request merged**
